src/components/common/CategoryTabBar/index.tsx

```diff
@@ -1,35 +1,63 @@
 import styled from '@emotion/styled';
+import { useEffect, useState } from 'react';
 
-import CategoryTabElement from '@/components/common/CategoryTabBar/CategoryTabElement';
+type TapWrapperProps = {
+  isActive: boolean;
+};
 
-export const TwoCategoryTabs = () => {
-  return (
-    <Wrapper>
-      <CategoryTabElement color="black" children="작품" />
-      <CategoryTabElement color="white" children="작가" />
-    </Wrapper>
-  );
+type CategoryTabBarProps = {
+  tabClick: (tab: string) => void;
+  tabState: string;
+  tabList: string[];
 };
 
-export const ThreeCategoryTabs = () => {
+const CategoryTabBar = ({ tabClick, tabState, tabList }: CategoryTabBarProps) => {
+  const [onActive, setOnActive] = useState('전체');
+
+  useEffect(() => {
+    setOnActive(tabState);
+  }, [tabState]);
+
   return (
     <Wrapper>
-      <CategoryTabElement color="black" children="통합" />
-      <CategoryTabElement color="white" children="작품" />
-      <CategoryTabElement color="white" children="작가" />
+      {tabList.map((category, index) => (
+        <TabWrapper
+          key={index}
+          isActive={onActive === category}
```

Comment:
tabState 그대로 `onActive` state에 들어가고 있는거 같은데 아래처럼 작성해보는건 어떨까요?

```suggestion
          isActive={tabState === category}
```